src/core/utils.rs

```diff
@@ -1,12 +1,24 @@
 use std::fmt::Debug;
 
 use quad_rand::compat::QuadRand;
-use rand::seq::SliceRandom;
+use rand::{
+    distributions::uniform::{SampleBorrow, SampleUniform},
+    seq::SliceRandom,
+    Rng,
+};
 
 pub fn zrng() -> impl rand::Rng {
     QuadRand
 }
 
+pub fn roll_dice<T: SampleUniform, B1, B2>(low: B1, high: B2) -> T
```

Comment:
Why different B1 and B2 types for each argument are required?

Reply:
> Why different B1 and B2 types for each argument are required?

Idk, but it's the signature of rand::Rng::gen_range(). I decided to use it for compatibility with its use cases.

Reply:
Ok, I'll simplify the signature then :)